ext/storage_sources/azure_store/azure_connection.cpp

```diff
@@ -95,3 +102,48 @@ azure_connection::get_object(const std::string &path) const
 {
     return 0;
 }
+
+int
+azure_connection::object_exists(const std::string &object_key, bool &exists) const
+{
+    exists = false;
+    std::string obj = _object_prefix + object_key;
+
+    auto list_blob_response = _azure_client.ListBlobs();
+
+    for (const auto blob_item : list_blob_response.Blobs) {
+        // Check if object exists.
+        if (blob_item.Name.compare(obj) == 0) {
+            // Check if object is deleted and has not been cleared by garbage collection.
+            if (blob_item.IsDeleted) {
+                return -1;
+            }
+            exists = true;
+        }
+    }
+    return 0;
+}
+
+int
+azure_connection::bucket_exists(bool &exists) const
```

Comment:
For this function, i believe there is a simpler way from following https://stackoverflow.com/questions/61128794/how-do-you-check-that-a-container-exists-in-azure-blob-storage-v12

Reply:
I also think we may not need the bucket_exists function and that we can directly perform this logic in the constructor.

Reply:
I came across that post when I was researching ways to implement `bucket_exists` but that functionality is for the .NET framework.